src/test/java/org/lsc/Ldap2LdapHookSyncTest.java

```diff
@@ -0,0 +1,261 @@
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *                  ==LICENSE NOTICE==
+ *
+ *               (c) 2008 - 2011 LSC Project
+ *         Sebastien Bahloul &lt;seb@lsc-project.org&gt;
+ *         Thomas Chemineau &lt;thomas@lsc-project.org&gt;
+ *         Jonathan Clarke &lt;jon@lsc-project.org&gt;
+ *         Remy-Christophe Schermesser &lt;rcs@lsc-project.org&gt;
+ ****************************************************************************
+ */
+package org.lsc;
+
+import com.fasterxml.jackson.databind.ObjectMapper; // For encoding object to JSON
+import com.fasterxml.jackson.databind.ObjectWriter;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.core.JsonFactory;
+import com.fasterxml.jackson.core.JsonParseException;
+import com.fasterxml.jackson.core.JsonParser;
+import java.io.StringWriter;
+import java.io.PrintWriter;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.lsc.configuration.LscConfiguration;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.io.File;
+import java.util.Scanner;
+
+
+/**
+ * This test case attempts to reproduce a ldap2ldap setup via SimpleSynchronize.
+ * It attempts to launch the tasks defined in src/test/resources/etc/lsc.xml:
+ * ldap2ldapHookTestCreate ldap2ldapHookTestUpdate ldap2ldapHookTestDelete
+ */
+public class Ldap2LdapHookSyncTest extends CommonLdapSyncTest {
+
+
+	@Before
+	public void setup() {
+		LscConfiguration.reset();
+		LscConfiguration.getInstance();
+		Assert.assertNotNull(LscConfiguration.getConnection("src-ldap"));
+		Assert.assertNotNull(LscConfiguration.getConnection("dst-ldap"));
+		reloadJndiConnections();
+	}
+
+	@Test
+	public final void testLdap2LdapJSONHookSyncTest() throws Exception {
+
+		// Declare the tasks to launch in the correct order
+		List<String> sync_tasks = Arrays.asList("ldap2ldapJSONHookTestCreate", "ldap2ldapJSONHookTestUpdate");
+		List<String> clean_tasks = Arrays.asList("ldap2ldapJSONHookTestDelete");
+
+		// perform the sync
+		launchSyncCleanTask(sync_tasks, false, true, false);
+		launchSyncCleanTask(clean_tasks, false, false, true);
+
+		// check the results of the synchronization
+		reloadJndiConnections();
+
+		ObjectMapper mapperCreatedEntry = new ObjectMapper();
+		JsonNode expectedCreatedEntry = mapperCreatedEntry.readTree("[ { \"attributeName\" : \"objectClass\", \"values\" : [ \"inetOrgPerson\", \"person\", \"top\" ], \"operation\" : \"ADD_VALUES\" }, { \"attributeName\" : \"cn\", \"values\" : [ \"CN0001-hook\" ], \"operation\" : \"ADD_VALUES\" }, { \"attributeName\" : \"sn\", \"values\" : [ \"CN0001-hook\" ], \"operation\" : \"ADD_VALUES\" } ]");
```

Comment:
could you add a binary attribute to your test? It will allow to check for regression regarding the handling of binary attributes in hooks.